tests/integration/kubernetes/tests_common.sh

```diff
@@ -85,68 +85,53 @@ auto_generate_policy_enabled() {
+is_coco_platform() {
+	case "${KATA_HYPERVISOR}" in
+		"qemu-tdx"|"qemu-snp"|"qemu-coco-dev")
+			return 0
+			;;
+		*)
+			return 1
+	esac
 }
 
-# adapt common policy settings for pod VMs using "shared_fs = virtio-fs" (https://github.com/kata-containers/kata-containers/issues/10189)
-adapt_common_policy_settings_for_virtio_fs() {
+adapt_common_policy_settings_for_non_coco() {
 	local settings_dir=$1
 
-	info "Adapting common policy settings for shared_fs=virtio-fs"
-	jq '.request_defaults.UpdateEphemeralMountsRequest = true' "${settings_dir}/genpolicy-settings.json" > temp.json && sudo mv temp.json "${settings_dir}/genpolicy-settings.json"
-	jq '.sandbox.storages += [{"driver":"virtio-fs","driver_options":[],"fs_group":null,"fstype":"virtiofs","mount_point":"/run/kata-containers/shared/containers/","options":[],"source":"kataShared"}]' \
-	"${settings_dir}/genpolicy-settings.json" > temp.json && sudo mv temp.json "${settings_dir}/genpolicy-settings.json"
-}
+	info "Adapting common policy settings from ${settings_dir} for non-CoCo guest"
 
-# adapt common policy settings for CBL-Mariner Hosts
-adapt_common_policy_settings_for_cbl_mariner() {
-	true
-}
+	# Using UpdateEphemeralMountsRequest - instead of CopyFileRequest.
+	jq '.request_defaults.UpdateEphemeralMountsRequest = true' "${settings_dir}/genpolicy-settings.json" > temp.json
+	sudo mv temp.json "${settings_dir}/genpolicy-settings.json"
 
-# adapt common policy settings for guest-pull Hosts
-# see issue https://github.com/kata-containers/kata-containers/issues/11162
-adapt_common_policy_settings_for_guest_pull() {
-	local settings_dir=$1
+	# Using a different path to container container root.
+	jq '.common.root_path = "/run/kata-containers/shared/containers/$(bundle-id)/rootfs"' "${settings_dir}/genpolicy-settings.json" > temp.json
+	sudo mv temp.json "${settings_dir}/genpolicy-settings.json"
 
-	info "Adapting common policy settings for guest-pull environment"
-	jq '.cluster_config.guest_pull = true' "${settings_dir}/genpolicy-settings.json" > temp.json && sudo mv temp.json "${settings_dir}/genpolicy-settings.json"
+	# Using CreateContainer Storage input structs for configMap & secret volumes - instead of using CopyFile like CoCo.
+	jq '.kata_config.enable_configmap_secret_storages = true' "${settings_dir}/genpolicy-settings.json" > temp.json
+	sudo mv temp.json "${settings_dir}/genpolicy-settings.json"
+
+	# Using watchable binds for configMap volumes - instead of CopyFileRequest.
+	jq '.volumes.configMap.mount_point = "^$(cpath)/watchable/$(bundle-id)-[a-z0-9]{16}-" | .volumes.configMap.driver = "watchable-bind"' \
+		"${settings_dir}/genpolicy-settings.json" > temp.json
+	sudo mv temp.json "${settings_dir}/genpolicy-settings.json"
```

Comment:
There was some condition on the runtime side that used watchable binds or non-watchable binds depending on the number of files in the CM, I believe. At least, I needed to relax the regexp at one point: https://github.com/edgelesssys/contrast/blob/main/packages/by-name/kata/kata-runtime/0012-genpolicy-allow-non-watchable-ConfigMaps.patch

Reply:
Reproducer is a config map with more than 8 entries: https://github.com/kata-containers/kata-containers/blob/b7af7f344b3f8cc60538e6f276c24ee1a8ad3dee/src/runtime/virtcontainers/mount.go#L408-L421

But we can leave it as is for now and add the new test case later - afaik, this is currently broken either way.